File: prob4.py
#!

def isPal(num):
	l = []
	r = []

	while num > 0:  
		l.append(num%10)
		r.append(num%10)
		num = num/10

	l.reverse()

	return l == r

answer = 0
found = False

for i in xrange(999, 100, -1):
	if found:
		break
	for j in xrange (999, 990, -1):
		if isPal(i*j):
			found = True
			answer = i*j
			break


print answer
print i 
#print j

File: prob4.cpp
#include <iostream>
using namespace std;

bool isPal(int num);
int DoingThings();

int main(){

	int answer = DoingThings();
	cout << answer << endl;
	
	
	return 0;
}

int DoingThings(){
	int biggestPal = 0;

	for (int i = 999; i >= 100 ; i--)
	{
		for (int j = 999; j >= 100 ; j--){
			if ((isPal(i*j)) && (i*j > biggestPal)){
			    biggestPal = i*j;
			}
		}
	}
	return biggestPal;
}

bool isPal(int num){

	int temp, firstDigit, revrs = 0;
	
	temp = num;
	
	while (temp > 0){
	    firstDigit = temp % 10;
	    temp = temp/10;
	    revrs = revrs*10 + firstDigit;
	}
	
	if (revrs == num) return true;
	else return false;

}